common/models/role.js

```diff
@@ -280,84 +318,87 @@ module.exports = function(Role) {
     debug('isInRole(): %s', role);
     context.debug();
 
```

Comment:
below: the easiest way without refactoring the whole method to turn it promisable while handling the callback style with early return implementation was to wrap the code in an anonymous self executed function. This way any `return` comprised inside this function does not return from the main method, but from the anonymous method, enabling the main method to return `cb.promise`

While the changes looks consequent they are in fact minimal.
Of course we could choose another option, but i feel this has a good compromise not to alter the existing flow